src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -548,4 +550,31 @@ public ICitizenData getRandomCitizen()
     {
         return (ICitizenData) citizens.values().toArray()[random.nextInt(citizens.values().size())];
     }
+
+    @Override
+    public void updateCitizenSleep(final boolean sleep)
+    {
+        this.isCitizensSleeping = sleep;
+    }
+
+    @Override
+    public void onCitizenSleep()
+    {
+
+
+        for (final ICitizenData citizenData : citizens.values())
+        {
+            if (!(citizenData.isAsleep() || citizenData.getJob() instanceof AbstractJobGuard))
+            {
+                return;
+            }
+        }
+
+        if (!this.isCitizensSleeping)
+        {
+            LanguageHandler.sendPlayersMessage(colony.getImportantMessageEntityPlayers(), ALL_CITIZENS_ARE_SLEEPING);
```

Comment:
getImportantMessageEntityPlayers will notify any officers even when not inside the colony, e.g. your friend who is officer in your colony aswell. Think you want to use the nonimportant ones